docs/user_guides/single_table/custom_constraints.rst

```diff
@@ -0,0 +1,239 @@
+.. _custom_constraints:
+
+Defining Custom Constraints
+===========================
+
+In some cases, the predefined constraints do not cover all your needs. 
+In such scenarios, you can use ``CustomConstraint`` to define your own 
+logic on how you would like to apply it to your data. There are three 
+main functions that you can create:
+
+- ``transform`` which is responsible for the forward pass when using ``transform`` strategy.
```

Comment:
Maybe also add what this method is meant to do. Like change the data in whatever way necessary to enforce constraint